packages/medusa/src/models/payment.ts

```diff
@@ -16,6 +16,7 @@ import { Order } from "./order"
 import { Swap } from "./swap"
 import { generateEntityId } from "../utils/generate-entity-id"
 
+@Index(["cart_id"], { where: "canceled_at IS NOT NULL" })
```

Comment:
question: Should this index include a config `{ where: ..., unique: true }`? To go along, we don't have a test verifying that multiple payments (active ones) can't exist on a cart. 
```suggestion
@Index(["cart_id"], { where: "canceled_at IS NOT NULL" })
```

Reply:
is that index correct? if it becomes uniq it would mean that we can't have multiple payment that are canceled, is that what we want?

Reply:
Ahh my bad!

Reply:
to only allow one active payment, we would probably want the same uniq index but with the opposite constraint

Reply:
Yes, that's what I was after

Reply:
In that case, do we want that new uniq index? If we don’t support multiple active payment then it makes sense 😀

Reply:
Yeah, I was asking because we currently do not allow more than one

Reply:
I will add it and the migration then

Reply:
Oh sh** i thought it was my pr sorry 😂

Reply:
Haha 😅

Reply:
Ahh yes - the idea with the index wasn't actually to enforce uniqueness, but I think it is a great idea to add that too! 

@adrien2p - would love if you could take over on this PR (you are already on your way I see 😄)

Reply:
And great point with adding a test case for double payments 👍

Reply:
😂 yeah sure (i really thought it was on one of the payment pr)